### DISCUSSION TITLE
end gcode after print / TFT override??

### DISCUSSION THREAD

**Discussion by demetan71** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2902] [Q&A]
Hi communauty.  Unfortunatly there is no manual to understand what TFT do.
It look to me that TFT bypass setting of marlin firmware present on the motherboard of the printer.

So i'm not shure if my problem is from the Marlin firmware or TFT issue

My main question, what start Gcode before print and end gcode after print does??  Cancel Gcode? Which gcode?? what it does??

In my slicer, my starting Gcode I have only G28  Where is starting ending gcode, Marlin firmware, TFT firmware, config.ini???

look at the video https://youtu.be/AuYFrXzrNqg

Bonjour communauté

Malheureusement il n'y a aucun manuel d'instruction à savoir ce que l'écran TFT fait.  Ce que les "feature" font vraiment.  Pour example, je me demande vraiment ce que fait la "feature" start gcode, end gcode et cancel gcode.

Je me demande ce que l'écran prend en charge, bypass ce qui est écrit dans le firmware.  J'aimerais comprendre son fonctionnement...!

Quand j'arrête une impression manuellement et que j'en repars une autre le homing se fait dans la mauvaise direction.

Lorsque je fini une impression, et que j'en repars une autre, le filament se rétracte tellement quelle sort parfois de l'extrudeur.

Regarder la vidéo https://youtu.be/AuYFrXzrNqg

**Comment by radek8** [comment]
Don't look for anything complicated in it.
Start gcode is executed before printing.
End gcode is executed after printing.
Cancel Code is triggered after canceling the print.
You can define these codes yourself.
If you don't need them, you can turn this feature off.

**Reply by demetan71** [reply]
Which starting gcode???  That's my question??  Were is the starting gcode and ending Gcode??  In Marlin firmware, In TFT firwmware??

That's is my main question???

Like, why it retract before my second print???  Where is writing in code to retract???  Why homing in the oposite ???

**Reply by radek8** [reply]
Gcode that you define yourself in the config.ini file in TFT

**Reply by demetan71** [reply]
Thank you!!  Appreciated

So the retrattion that my printer does is somewhere in the motherboard firmware??  In which file??

And homing in wrong direction when I stop a print, maybe is when a print finshised normaly it put it in relative or absolute....  And this is in the firmware of the motherboard, but where??  :)

**Reply by radek8** [reply]
The pull can be done by Marlin
Check Configuration.h
parameter:
#define Z_PROBE_END_SCRIPT

**Reply by radek8** [reply]
What parameter do you have there?

**Reply by demetan71** [reply]
This line is commented //#define Z_PROBE_END_SCRIPT "G1 Z10 F12000\nG1 X15 Y330\nG1 Z0.5\nG1 Z10"

My homing position in wrong direction that I really want to understand, and wanted to know where in the firmware is set retact filament.

So as what you said to me, when I stop a print, it have no code at all....  I'm really confused

**Comment by radek8** [comment]
Don't you have the retraction set in the slicer as an end code?

**Reply by demetan71** [reply]
oh!  I didn't use always the same slicer, I use Ideamaker, and Cura, I just verified it nad in Cura, yes I have this setting in Cura, at the begining.  But only 3mm, but it retract a lot more, and like I said he does this only when I stop manually a print before it's end and starting another....   For the homing in the wrong direction, I tough it is related....!!???

**Comment by radek8** [comment]
Does the printer move in the right direction when printing and in the opposite direction when manually moving?
Does it misbehave equally in the Marlin screen and in the touch menu?

**Reply by demetan71** [reply]
Thank you again, it's appreciated, but give me time to investigate....  I don't whant to said bulls.....!! ;)

**Comment by radek8** [comment]
Do I understand you correctly?
You do HOME before printing and it goes in the right direction.
Then you start printing and the head moves in the right direction.

Do you cancel the print and the Y axis moves fine and the X axis goes the other way when you do HOME?

**Reply by demetan71** [reply]
Ok, let my start again!!!  (maybe it's me, I'm a french candian so!!)

I start a print, I realize that something going wrong, part won't stick or whatever, so I stop the print on the LCD in touch mode, i didn't use marlin mode.

And the homing, and the too much retraction happend when I start another print.  So that's why I tought is related whit an end gcode or start gcode.

When I open the printer, my first print everything is ok.

To answer your question, when it homing in wrong direction I hit the turn off button as fast as I can because it crash at the end of rail (no limit switch at right....

But next time, instead of starting a new print I will test if when I try to move manually on the lcd if it goes in the good direction.

So first print, everything ok

**Reply by radek8** [reply]
Send the files to Marlin
Configuration.h
Configutaton_adv.h

TFT file
config.ini
Thank you

**Reply by demetan71** [reply]
[Configuration.txt](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/14466878/Configuration.txt)
[Configuration_adv.txt](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/14466879/Configuration_adv.txt)

**Reply by radek8** [reply]
Are these the files you compiled your own Marlin with?
Also send the Config.ini that you uploaded to the TFT display.

**Reply by demetan71** [reply]
Visual studio, I know, I think that's not the best, sometime I had trouble and realise that I have to close Visual studio nd reopen it.....  But I never been able to compile with arduino ide my SKR 1.4 board, he didn't recognize environement....
[config.txt](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/14469688/config.txt)

**Reply by radek8** [reply]
I was looking at your configuration files.
I haven't noticed anything that I think could be causing your problems.

**Reply by demetan71** [reply]
Thank you
